Basic SCH implementation

xiankai opened the pull request:
Used the SMN implementation as a reference to add 3 modules:
- Aetherflow
  - Changed implementation to check against a 45 second cooldown
- DoTs
  - Replace with SCH-specific DoTs (Miasma, Bio II)
- Shadow Flare
  - Copied as is

ackwell commented on src/parser/jobs/sch/Aetherflow.js:
```diff
@@ -0,0 +1,86 @@
+		}))
+	}
+
+	getQuickenedAetherflowOnCooldown() {
```
Should keep it using `this.cooldowns.getTimeOnCooldown`.

You can call `this.cooldowns.reduceCooldown` when players cast CDR skills, it'll automatically calculate the correct time on cooldown once that's implemented.

ackwell commented on src/parser/jobs/sch/About.js:
```diff
@@ -0,0 +1,24 @@
+import CoreAbout from 'parser/core/modules/About'
+
+export default class About extends CoreAbout {
+	description = <Fragment>
```
This file needs to be updated to represent the state of the SCH module group.

ackwell commented on src/parser/jobs/sch/Aetherflow.js:
```diff
@@ -0,0 +1,86 @@
+				new Requirement({
+					name: <Fragment><ActionLink {...ACTIONS.AETHERFLOW} /> cooldown uptime</Fragment>,
+					percent: (this.getQuickenedAetherflowOnCooldown() / (this.parser.fightDuration - FIRST_FLOW_TIMESTAMP)) * 100,
+				}),
```
I'd add a separate Requirement here to deal with using CDR flow actions ineffectively. Uptime and effective usage are sort of two separate things.

ackwell commented on src/parser/jobs/sch/DoTs.js:
```diff
@@ -0,0 +1,134 @@
+		this.suggestions.add(new Suggestion({
+			icon: ACTIONS.BIO.icon,
+			content: <Fragment>
+				Avoid refreshing DoTs significantly before their expiration, except when at the end of the fight. Unnecessary refreshes risk overwriting buff snapshots, and increase the frequency you&apos;ll need to hardcast your DoTs.
```
This suggestion is primarily worded for SMN, where you want to keep DoTs up with Tri-Disaster as much as possible. Might need some tweaks.

xiankai commented:
After our discussion, I've addressed the following issues:
- Use in-built CDR handling
- Compare total aetherflow casts/possible aetherflow casts (added under requirement, or should it be separated out into a suggestion?)
- Updated contribution
- Text fixes (for DoT overwriting and aetherflow description)

And what is hopefully a more holistic overview of aetherflow usage (could add suggestions or highlight problematic rows later):

![image](https://user-images.githubusercontent.com/1176845/43090942-61e45740-8edb-11e8-9df6-301aaa8b3e27.png)

ackwell commented:
Thanks for those changes, and that table looks great, nice job.

As I've just posted in `#development`, I'll be requesting a new description in the `about` module for the next PR, please and thankee.

Also for future PRs, the code generating the table could probably do with a bit of simplification - might be worth pulling that block outside the JSX and _maybe_ into a few functions depending on feasibility, just to break it up and make it a bit more digestible.
